catalog/solr/catalog-solr-external-provider/src/main/java/ddf/catalog/solr/external/SolrHttpCatalogProvider.java

```diff
@@ -263,21 +267,22 @@ private void updateServer() {
     }
 
     private CatalogProvider getProvider() {
-        if (isServerUp(getServer())) {
-            if (resolver == null) {
-                provider = new SolrCatalogProvider(getServer(),
-                        filterAdapter,
-                        solrFilterDelegateFactory);
-            } else {
-                provider = new SolrCatalogProvider(getServer(),
-                        filterAdapter,
-                        solrFilterDelegateFactory,
-                        resolver);
+        if (firstUse) {
```

Comment:
Well, the problem with reverting to the `firstUse` pattern is that this no longer fixes the bug(s) it was supposed to fix. This logic doesn't configure the catalog provider. It sounds like we just need to check if the provider has already been initialized in the `isServerUp()` check at the top.

I can work with you and @NGoss on this more when everyone gets back from vacation. I don't think it's ready for a PR as-is, so you might want to close the one that's currently out.